### ISSUE TITLE
Secret permissions too wide ?

### ISSUE THREAD

**Issue by aelbarkani** [redhat-cop/group-sync-operator#64]
It seems the operator has "get, watch, list" on the secrets of all namespaces. I assume that it's because it's possible to use secrets from any namespace. But is that necessary ? Wouldn't be wiser to allow only to read secrets from its own namespace ?

**Comment by sabre1041** [comment]
This decision was to enable being able to reference credentials deployed/found in different namespaces. 

A roadmap item is to move this operator to a cluster scoped operator which will require permissions at a cluster level to read from any namespace

**Comment by aelbarkani** [comment]
Yes, but I mean, why this requirement would be necessary ? In my humble opinion, reading a secret from own namespace is more than enough I think.
As for the cluster-scoped part, group sync operator needs to be deployed only by a cluster administrator and only once per cluster. So a namespaced operator looks fine to me.

**Comment by sabre1041** [comment]
The idea is that you might want to reuse a secret found in another namespace (for example `openshift-config` where there is a hard requirement that secrets be located there). Instead of maintaining authentication credentials in one namespace and authorization credentials in another, you could reuse the same, but deploy the operator in a separate namespace

**Comment by aelbarkani** [comment]
I see. Well, in that case, wouldn't be enough to have a simple Role and RoleBinding on that specific namespace (openshift-config for example) ? Unless it is expected for this operator to become a ClusterOperator (or maybe part of the Authentication ClusterOperator).

**Comment by sabre1041** [comment]
Cluster scoped will most likely occur as part of the `v0.0.10` release. `v0.0.9` is in review and should be in Operator Hub soon to resolve #61

**Comment by aelbarkani** [comment]
I may be wrong but Cluster scoped operators means that anyone in a multi-tenant cluster can instantiate a GroupSync. Is that really what is intended ? In my opinion it's a security issue. You can perfectly have namespace scoped operators that manage a cluster-wide aspect.

cc @raffaelespazzoli

**Comment by sabre1041** [comment]
The first step is to move the CRD to be cluster scoped instead of namespaced scoped. That is the primary impetus behind the change.

After that, considerations on the scope of the operator itself will be determined. Your input and concern is greatly appreciated!
